paymentsheet/src/main/java/com/stripe/android/customersheet/StripeCustomerAdapter.kt

```diff
@@ -140,12 +140,24 @@ internal class StripeCustomerAdapter @Inject constructor(
                     cachedCustomerEphemeralKey.date
                 )
             }?.result ?: run {
+                val ephemeralKey = customerEphemeralKeyProvider.provideCustomerEphemeralKey()
+
+                ephemeralKey.getOrNull()?.let {
+                    customerRepository.retrieveCustomer(
+                        customerId = it.customerId,
+                        ephemeralKeySecret = it.ephemeralKey,
+                    ) ?: return@run CustomerAdapter.Result.failure(
+                        cause = IllegalStateException("Invalid customer"),
```

Comment:
Is `IllegalStateException` the right exception to return? Should we create our own exception?